src/Builder/Creator/MethodCreator.php

```diff
@@ -0,0 +1,233 @@
+
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\NodeTraverser;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+use SprykerSdk\Integrator\Builder\Exception\LiteralValueParsingException;
+use SprykerSdk\Integrator\Builder\Visitor\AddMethodVisitor;
+use SprykerSdk\Integrator\Transfer\ClassInformationTransfer;
+
+class MethodCreator extends AbstractMethodCreator implements MethodCreatorInterface
+{
+    /**
+     * @var int
+     */
+    protected const SINGLE_TREE_RETURN_STATEMENT_COUNT_ELEMENTS = 1;
+
+    /**
+     * @var \SprykerSdk\Integrator\Builder\Creator\MethodStatementsCreatorInterface
+     */
+    protected $methodStatementsCreator;
+
+    /**
+     * @var \SprykerSdk\Integrator\Builder\Creator\MethodDocBlockCreatorInterface
+     */
+    protected $methodDocBlockCreator;
+
+    /**
+     * @var \SprykerSdk\Integrator\Builder\Creator\MethodReturnTypeCreatorInterface
+     */
+    protected $methodReturnTypeCreator;
+
+    /**
+     * @var \PhpParser\ParserFactory
+     */
+    protected $parserFactory;
+
+    /**
+     * @param \SprykerSdk\Integrator\Builder\Creator\MethodStatementsCreatorInterface $methodStatementsCreator
+     * @param \SprykerSdk\Integrator\Builder\Creator\MethodDocBlockCreatorInterface $methodDocBlockCreator
+     * @param \SprykerSdk\Integrator\Builder\Creator\MethodReturnTypeCreatorInterface $methodReturnTypeCreator
+     * @param \PhpParser\ParserFactory $parserFactory
+     */
+    public function __construct(
+        MethodStatementsCreatorInterface $methodStatementsCreator,
+        MethodDocBlockCreatorInterface $methodDocBlockCreator,
+        MethodReturnTypeCreatorInterface $methodReturnTypeCreator,
+        ParserFactory $parserFactory
+    ) {
+        $this->methodStatementsCreator = $methodStatementsCreator;
+        $this->methodDocBlockCreator = $methodDocBlockCreator;
+        $this->methodReturnTypeCreator = $methodReturnTypeCreator;
+        $this->parserFactory = $parserFactory;
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param array|string|float|int|bool|null $value
+     *
+     * @throws \SprykerSdk\Integrator\Builder\Exception\LiteralValueParsingException
+     *
+     * @return array<array-key, \PhpParser\Node\Stmt>
+     */
+    public function createMethodBody(ClassInformationTransfer $classInformationTransfer, $value): array
+    {
+        if (is_array($value)) {
+            return $this->createReturnArrayStatement($classInformationTransfer, $value);
+        }
+
+        $preparedValue = $this->createPreparedValueForParser($value);
+        $tree = $this->parserFactory->create(ParserFactory::PREFER_PHP7)->parse($preparedValue);
+        if (!$tree) {
+            throw new LiteralValueParsingException(sprintf('Value is not valid PHP code: `%s`', $value));
```

Comment:
but it parses not only literal values as I can see

Reply:
Anyway if we have invalid expression, we can't continue our logic

Reply:
Yes, but what I mean, we throw `LiteralValueParsingException` even if the error is not related to the literal value parsing